models/stock_move_line.py

```diff
@@ -372,11 +372,22 @@ def _split(self):
                         {'product_uom_qty': done_to_keep,
                          'qty_done': qty_done,
                          })
-            self.ordered_qty= ordered_qty
+            self.ordered_qty = ordered_qty
             res = new_ml
 
         return res
 
+
+    def _get_all_products_quantities(self):
+        '''This function computes the different product quantities for the given move_lines
+        '''
+        res = {}
+        for move_line in self:
+            if move_line.product_id not in res:
```

Comment:
Using defaultdict you could remove this if.

Reply:
fair enough